tests/namespace_usage_test.py

```diff
@@ -0,0 +1,52 @@
+import ast
+import tokenize
+from io import StringIO
+
+import pytest
+
+from pandas_dev_flaker.__main__ import run
+
+
+def results(s):
+    return {
+        "{}:{}: {}".format(*r)
+        for r in run(
+            ast.parse(s),
+            list(tokenize.generate_tokens(StringIO(s).readline)),
+        )
+    }
+
+
+@pytest.mark.parametrize(
+    "source",
+    (
+        pytest.param(
+            "from pandas import Categorical\n"
+            "\n"
+            "cat1 = Categorical()\n"
+            "cat2 = Categorical()\n",
+            id="Both are imported from pandas",
+        ),
+    ),
+)
+def test_noop(source):
+    assert not results(source)
+
+
+@pytest.mark.parametrize(
+    "source, expected",
+    (
+        pytest.param(
+            "from pandas import Categorical\n"
+            "import pandas as pd\n"
+            "\n"
+            "cat1 = Categorical()\n"
+            "cat2 = pd.Categorical()\n",
+            "5:7: PDF011 found both 'pd.foo' and 'foo' in the same file",
+            id="One is accessed from pd, the other is imported from pandas",
+        ),
+    ),
```

Comment:
Awesome!

Can we add an extra test in which we have
```
import pandas
from pandas import Categorical

cat1 = Categorical()
cat2 = pd.Categorical()
```
?

Reply:
if it doesn't pass, that's OK, let me know and I'll open a new issue for it and we can merge this one as it is

Reply:
Thanks! Just added it.